### ISSUE TITLE
Problem with a nozzle at the boundary using PERIODIC FLOW ONLY boundary conditions

### ISSUE THREAD

**Issue by rakyuv** [firemodels/fds#13632]
Hi,

I'd like to simulate methane release from a small vent at a boundary of the domain, and use "PERIODIC FLOW ONLY" boundary condition on both horizontal directions. This simulation ([testNoTerrain.txt](https://github.com/user-attachments/files/17542116/testNoTerrain.txt)) runs correctly. 
However, when I add terrain data using GEOM ([testTerrainGEOM.zip](https://github.com/user-attachments/files/17542133/testTerrainGEOM.zip)), it crashes. On the other hand, when I use terrain with GEOM and "OPEN" boundary conditions ([testTerrainGEOM_OPEN.zip](https://github.com/user-attachments/files/17542800/testTerrainGEOM_OPEN.zip)), the code runs. 

This gives me only three options. To run the simulation and:
(1) Use OPEN boundary conditions, with terrain from GEOM.
(2) Use PERIODIC FLOW ONLY boundary condition, and use it without terrain from GEOM.
(3) Use PERIODIC FLOW ONLY boundary condition and use with Terrain from OBST. 

I'd like to run the simulation with VENT on the boundary and be able to have terrain from GEOM and PERIODIC FLOW ONLY boundary condition in both horizontal directions. 

**Desktop (please complete the following information):**
 - OS: Linux

Any help would be greatly appreciated

Thanks in advance

**Comment by marcosvanella** [comment]
The issue here is that the terrain and hence the grid is not periodic. When setting up the ghost cells to fill the periodicity condition FDS sees that these don't match, as the grid is not periodic. I'll work on making this a graceful exit.
For the time being you will need to use open boundaries for general terrains.

**Comment by rakyuv** [comment]
Thank you @marcosvanella for your answer.
I thought about this problem and I found the following ideas:
1. Change the coordinate system, and therefore change the grid cells to to follow the terrain. I have seen it in WRF and also ERA5 datasets which are usually terrain following. 
2. If there is a way to extend the domain artifically until the terrain profile is the same on both the periodic boundaries. I had worked on this idea in the past, to add periodicity to a non-periodic data for pushing the data into Fourier space. 

For the moment, I'll close the issue. Feel free to update me once you have some solution to the problem. 

Thanks in advance.
